src/text/non_blocking/raw_text_reader.rs

```diff
@@ -1261,4 +1408,91 @@ mod reader_tests {
         assert!(result.is_err());
         Ok(())
     }
+
+    // Test stepping out at the top level with incomplete error during step_out call.
+    #[test]
+    fn resume_step_out() -> IonResult<()> {
+        // The spacing on this is important, since we need to know where (approximately) the
+        // incomplete error occurs.
+        let source = r#"{field:{another_field:{foo:"We should fail somewhere in this string.."}}}"#;
+
+        let mut reader = RawTextReader::new(source[..47].as_bytes().to_owned());
```

Comment:
Could you add a comment showing where 47 is in the input? I see it's somewhere in the string, but knowing exactly where would help me mentally run the unit test.